### PR TITLE
Added Windows CMake setup info

### PR DESCRIPTION
**Pull request by BlackHoleFox**
Changes proposed in this pull request:
 -  Added Windows CMake development information as the config added support for it

Your great patch is much appreciated. We are considering to apply your patch into the SoftEther VPN main tree.

SoftEther VPN Patch Acceptance Policy:
http://www.softether.org/5-download/src/9.patch

You have two options which are described on the above policy.
Could you please choose either option 1 or 2, and specify it clearly on the reply?

- 1

I also agree to my work being licensed under a non GPL license anytime in the future.

I'm planning to sometime soon (hopefully) create a GitHub Wiki entry for development on Windows, though that will be in the future.

Feel free to point out any grammar/spelling mistakes and I'll be happy to correct them.

### CODE REVIEW

File: src/BUILD_WINDOWS.md [hunk 1]
```diff
@@ -40,7 +40,43 @@ If using anything else other than Visual Studio 2008 for development, your code
...
+
+There are several methods for using CMake but the easiest by far is through Visual Studio 2017 by importing the CMake project directly
+into it. So that is what will be described below.
+
+Requirements:
+
+1. Download Visual Studio 2017 (Community Edition is fine).
+2. During install, make sure to check "Desktop development with C++" under "Workloads".
+3. Click on individual components and scroll until you see "Visual C++ tools for CMake" under the compilers section. Make sure this is checked.
+4. Continue scrolling down until you reach "SDKs, libraries, and frameworks". Under this tab you want to check to options:
```
Comment (chipitsine) [inline src/BUILD_WINDOWS.md]: can we add command line installation instrruction ?

something based on workload is

https://docs.microsoft.com/en-us/visualstudio/install/workload-and-component-ids?view=vs-2017

File: src/BUILD_WINDOWS.md [hunk 2]
```diff
@@ -40,7 +40,43 @@ If using anything else other than Visual Studio 2008 for development, your code
...
+
+Visual Studio will proceed to start the CMake configuration process and once its finished, you can simply go to toolbar and click `CMake -> Build All`.
+
+Once it has finished, hopefully with no errors, look in the newly created `/build` directory in the project's folder. Inside are the development versions
+of all the SoftEtherVPN components.
+
+Congrats, you now have a complete CMake development environment for SoftEtherVPN on Windows, enjoy and happy contributing!
+
+Download Links:
+- (One of a few) OpenSSL Windows Installer: https://slproweb.com/products/Win32OpenSSL.html
```
Comment (chipitsine) [inline src/BUILD_WINDOWS.md]: can we use nuget here ?

File: src/BUILD_WINDOWS.md (hunk 1 above)
Comment (chipitsine) [reply src/BUILD_WINDOWS.md]: I mean, when VS2017 is installed, you can add workloads to existing installation

Comment (BlackHoleFox) [reply src/BUILD_WINDOWS.md]: I definitely could add something about just adding it as another workload the process is dead identical as VS2017 installs and modifies through the Visual Studio Installer application.

I would have to research the CLI instructions, as I have never touched them before. Do standard Windows developers use them that frequently?

File: src/BUILD_WINDOWS.md (hunk 2 above)
Comment (BlackHoleFox) [reply src/BUILD_WINDOWS.md]: I don't think so. Nuget is for C#/.NET projects. Everything I read says you need to download and have the openssl libs somewhere that your build can find them.

File: src/BUILD_WINDOWS.md (hunk 1 above)
Comment (chipitsine) [reply src/BUILD_WINDOWS.md]: we use things like that in automated build agent setup. if you think developers do not use it frequently, let's ignore that

Comment (BlackHoleFox) [reply src/BUILD_WINDOWS.md]: At the moment, the AppVeyor has a Visual Studio 2017 build process, so I assume its already setup in some way?

File: src/BUILD_WINDOWS.md (hunk 2 above)
Comment (davidebeatrici) [reply src/BUILD_WINDOWS.md]: Installing OpenSSL is not required, as we provide the static libraries in the repository:
https://github.com/SoftEtherVPN/SoftEtherVPN/blob/080edcdb8155ebaf110dfab1bd747a7fd0728203/src/Mayaqua/CMakeLists.txt#L25-L45

Comment (BlackHoleFox) [reply src/BUILD_WINDOWS.md]: @davidebeatrici Is there a way to tell CMake this by default? It didn't pick them up automatically.

Comment (davidebeatrici) [reply src/BUILD_WINDOWS.md]: It should find the libraries automatically, as we specify the path inside the repository.

Comment (BlackHoleFox) [reply src/BUILD_WINDOWS.md]: Odd. Removed the ones I had separately and it still worked. I'll just put it up to my system being weird. Incoming commit removing that.

Comment (davidebeatrici) [reply src/BUILD_WINDOWS.md]: No problem.

Comment (BlackHoleFox) [reply src/BUILD_WINDOWS.md]: Should be fine now. CI passed for the changes removing the OpenSSL info

Comment (davidebeatrici) [reply src/BUILD_WINDOWS.md]: Could you squash the commits, please?

Comment (BlackHoleFox) [reply src/BUILD_WINDOWS.md]: @davidebeatrici So far I have made 3 messes out of my repo. Can you link me to a good way to squash commits?

Comment (davidebeatrici) [reply src/BUILD_WINDOWS.md]: Sure: https://stackoverflow.com/a/5189600

Comment (BlackHoleFox) [reply src/BUILD_WINDOWS.md]: I think that worked?

Comment (davidebeatrici) [reply src/BUILD_WINDOWS.md]: Indeed!

**Comment by davidebeatrici** [comment]
Thank you very much for your contribution!

**Comment by BlackHoleFox** [comment]
No problem. Wiki entry soon(TM)

**Pull request merged**